### ISSUE TITLE
Autosubmit bot landed a commit while the tree was red.

### ISSUE THREAD

**Issue by jonahwilliams** [flutter/flutter#131281]
https://github.com/flutter/flutter/pull/129688 was submitted by the autosubmit bot while the tree was red. Its not clear if this was a github status race or another issues with the configuration.

The previous commit was landed on red to fix the build.

**Comment by ricardoamador** [comment]
From the point of view of autosubmit the tree-status check was green.

![image](https://github.com/flutter/flutter/assets/32242716/688023e2-8725-4aaa-9948-42e8a81c169a)

**Comment by ricardoamador** [comment]
They may not run at the same time either. This is entirely possible that the tree was green briefly at Autosubmit constantly checks this.

**Comment by keyonghan** [comment]
Seems this happened on Engine tree as well. The recent two commits were landed ignoring the red. I manually marked the task as `Failed`, so the tree should have been remaining closed all the time?

<img width="1223" alt="Screenshot 2023-07-25 at 2 03 56 PM" src="https://github.com/flutter/flutter/assets/54558023/44358439-2192-434a-a842-a4962b0e1902">

**Comment by ricardoamador** [comment]
Doesn't this depend on the update cycle though?

**Comment by ricardoamador** [comment]
Looking into the specific issue mentioned I was able to find the following:

Autosubmit validated the status of https://github.com/flutter/flutter/pull/129688 as reported in the log at 2023-07-25T18:24:41.633100220Z which is 11:24:41.633 pacific time.

![image](https://github.com/flutter/flutter/assets/32242716/5ddbd84f-eb52-498b-b832-f6803a46a468)

The pull request was merged 2 seconds later at "2023-07-25T18:24:43Z", which is 11:24:43 pacific time as reported by Github.

![image](https://github.com/flutter/flutter/assets/32242716/a0684dea-7d9a-4de8-90e4-d6803ce8794e)

In this case it is highly doubtful autosubmit merged the pull request on a red tree as reported by infra services.
